Lab5/GABDemo/Services/ImageAnalyzer.cs

```diff
@@ -20,18 +20,16 @@ public class ImageAnalyzer : IImageAnalyzer
 
         public ImageAnalyzer(ComputerVisionClient computerVision)
         {
-            _computerVision = computerVision;
+            _computerVision = computerVision ?? throw new ArgumentNullException(nameof(computerVision));
         }
 
-        public async Task<ImageAnalysis> AnalyzeAsync(string imageUrl)
+        public Task<ImageAnalysis> AnalyzeAsync(string imageUrl)
         {
             if (!Uri.IsWellFormedUriString(imageUrl, UriKind.Absolute))
             {
-                throw new Exception("Invalid remoteImageUrl: {imageUrl}");
+                throw new Exception($"Invalid remoteImageUrl: {imageUrl}");
             }
-
-            ImageAnalysis analysis = await _computerVision.AnalyzeImageAsync(imageUrl, Features);
-            return analysis;
+            return _computerVision.AnalyzeImageAsync(imageUrl, Features);
```

Comment:
Personally, when I review, I prefer having the return statement split as it's easier for someone who wants to debug to put a breakpoint and see the properties of the variable. Refer to global comment for me feedback

Reply:
True enough (I think I said that to someone last week), I'll revert this to a "two-step return statement".